### PR TITLE
[Feat] Add more augmentations

### PR DESCRIPTION
**Pull request by edgarriba**
### Description

Implements:
- `kornia.enhance.invert`
- `kornia.enhance.Invert`
- `kornia.augmentation.RandomInvert`
- `kornia.augmentation.RandomChannelShuffle`
- `kornia.augmentation.RandomGaussianNoise`

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/augmentation/augmentation.py [hunk 1]
```diff
@@ -1369,3 +1370,46 @@ def generate_parameters(self, batch_shape: torch.Size) -> Dict[str, torch.Tensor
...
+                  [0.6816, 0.9152, 0.3971, 0.8742, 0.4194]]]])
+    """
+
+    def __init__(self,
+                 max_val: torch.Tensor = torch.tensor(1.),
+                 return_transform: bool = False,
+                 same_on_batch: bool = False,
+                 p: float = 0.5) -> None:
+        super(RandomInvert, self).__init__(
+            p=p, return_transform=return_transform, same_on_batch=same_on_batch, p_batch=1.)
```
Comment (edgarriba) [inline kornia/augmentation/augmentation.py:1405]: @shijianjian wasn't unsure about the `p_batch` params - just copied from GaussianBlur

Comment (shijianjian) [reply kornia/augmentation/augmentation.py:1405]: ```p_batch``` is used to control the probability batch-wisely, as there is no ragged tensor supported for cropping function. Thus, cropping happens either the whole batch or nothing happens at all. ```p_batch``` shall be 1 under this context.

File: kornia/enhance/adjust.py [hunk 2]
```diff
@@ -943,3 +971,33 @@ def __init__(self, brightness_factor: Union[float, torch.Tensor]) -> None:
...
+        torch.Size([1, 2, 3, 4, 4])
+
+        >>> img = torch.rand(1, 3, 4, 4)
+        >>> Invert(torch.tensor([[[[1.]]]]))(img).shape
+        torch.Size([1, 3, 4, 4])
+    """
+
+    def __init__(self, max_val: torch.Tensor = torch.tensor(1.)) -> None:
+        super(Invert, self).__init__()
+        self.max_val = max_val
```
Comment (shijianjian) [inline kornia/enhance/adjust.py:1000]: Currently alright. I am thinking to have:

```python
if not isinstance(max_val, nn.Parameter): 
    self.register_buffer("max_val", max_val)
else:
    self.max_val = max_val
```

To make it more appropriate.

File: kornia/enhance/adjust.py [hunk 3]
```diff
@@ -747,6 +749,32 @@ def equalize3d(input: torch.Tensor) -> torch.Tensor:
     return torch.stack(res)
 
 
+def invert(input: torch.Tensor, max_val: torch.Tensor = torch.tensor(1.0)) -> torch.Tensor:
```
Comment (shijianjian) [inline kornia/enhance/adjust.py:752]: This currently supports any shaped tensors. Shall we restrict it to be at least 2D?

File: kornia/enhance/adjust.py [hunk 4]
```diff
@@ -747,6 +749,32 @@ def equalize3d(input: torch.Tensor) -> torch.Tensor:
...
+        >>> invert(img, torch.tensor(255.)).shape
+        torch.Size([1, 2, 3, 4, 4])
+
+        >>> img = torch.rand(1, 3, 4, 4)
+        >>> invert(img, torch.tensor([[[[1.]]]])).shape
+        torch.Size([1, 3, 4, 4])
+    """
+    assert isinstance(input, torch.Tensor), f"Input is not a torch.Tensor. Got: {type(input)}"
+    assert isinstance(max_val, torch.Tensor), f"max_val is not a torch.Tensor. Got: {type(max_val)}"
+    return max_val.to(input) - input
```
Comment (shijianjian) [inline kornia/enhance/adjust.py:775]: I think we shall cast our internal used tensors, but not the input tensors. 
- If the user input two tensors in different devices, let it crash. 
- If the user input two tensors in different types, we cast dtype only.

File: kornia/augmentation/augmentation.py [hunk 5]
```diff
@@ -1369,3 +1370,46 @@ def generate_parameters(self, batch_shape: torch.Size) -> Dict[str, torch.Tensor
...
+        >>> inv(img)
+        tensor([[[[0.4963, 0.7682, 0.0885, 0.1320, 0.3074],
+                  [0.6341, 0.4901, 0.8964, 0.4556, 0.6323],
+                  [0.3489, 0.4017, 0.0223, 0.1689, 0.2939],
+                  [0.5185, 0.6977, 0.8000, 0.1610, 0.2823],
+                  [0.6816, 0.9152, 0.3971, 0.8742, 0.4194]]]])
+    """
+
+    def __init__(self,
+                 max_val: torch.Tensor = torch.tensor(1.),
```
Comment (shijianjian) [inline kornia/augmentation/augmentation.py:1400]: Shouldn't be a Union type? ```int, float, tensor```?

File: kornia/enhance/adjust.py (hunk 4 above)
Comment (ducha-aiki) [reply kornia/enhance/adjust.py:775]: +1

File: kornia/enhance/adjust.py (hunk 2 above)
Comment (ducha-aiki) [reply kornia/enhance/adjust.py:1000]: +1

File: kornia/augmentation/augmentation.py (hunk 5 above)
Comment (edgarriba) [reply kornia/augmentation/augmentation.py:1400]: if any users asks we can implement later
